### PR TITLE
feat/AB#104720_EIS-implement-add-record-action-of-dashboard-buttons

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

feat: add addRecord form fields integration for button action edit modal 
feat: add needed queries for the reason above 
refactor: edit button action modal code to be more accessible and easy to follow feat: add new related translations 
feat: update button action type model

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/104720)
- [Link to back-end branch](https://github.com/ReliefApplications/ems-backend/pull/1139)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

Please include screenshots of this change. If this issue is only back-end related, and does not involve any visual change of the platform, you can skip this part.

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by unai-reliefapp** [comment]
@AntoineRelief current flow for add record button, please confirm me if there is anything missing(besides maybe displayed text improvements ofc)
Button creation
https://github.com/user-attachments/assets/d540468d-74f7-454f-974a-011cfa6bca23
Selected record in dashboard context 
https://github.com/user-attachments/assets/82057ae9-29ea-47ac-ae32-157465e472a2
No record selected in dashboard context(new record modal)
https://github.com/user-attachments/assets/7b752956-7120-4284-9a2c-844826b4f5ff

User can select any other resource, templates, fields, etc, but by default if there is any set in the given dashboard context, would be automatically set on new button form creation

KR, Unai

File: apps/back-office/src/app/dashboard/pages/dashboard/components/edit-button-action-modal/graphql/queries.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { gql } from 'apollo-angular';
+
+/** Graphql request for getting resource data */
+export const GET_RESOURCE = gql`
+  query GetGridResourceMeta($resource: ID!) {
```
Comment (AntoineRelief) [inline apps/back-office/src/app/dashboard/pages/dashboard/components/edit-button-action-modal/graphql/queries.ts]: @unai-reliefapp 

be aware when copying that sometimes it's out of context
**GetGridResourceMeta** doesn't work there

**Comment by AntoineRelief** [comment]
@unai-reliefapp 

I'll take the review & do the according changes
I didn't notice it with your videos, but you took the wrong way in fact

- addRecord modal is only about adding new records, it's not about editing the current one. So opening the modal injecting current data was wrong
- the resource fields must come from the current resource of the context, not from the fields of the resource you selected in the modal

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp
> 
> I'll take the review & do the according changes I didn't notice it with your videos, but you took the wrong way in fact
> 
> * addRecord modal is only about adding new records, it's not about editing the current one. So opening the modal injecting current data was wrong
> * the resource fields must come from the current resource of the context, not from the fields of the resource you selected in the modal

So the add record modal should open a form using the selected resource and template in the new button form. And how those fields integrate in the open form modal?

**Comment by AntoineRelief** [comment]
@unai-reliefapp 

Updated & merged

a few changes:
- dashboard is directly passed from modal to modal instead of relying on the service to pass it. I don't really like this approach
- the addRecord now works as expected
- I added a callback on addRecord, so when the record is generated & you have a record in the current view, the current record can be updated. I'll let you check again the mockup to see how it is supposed to work

**Pull request merged**